### PR TITLE
Task/remove semantic UI (part 1)

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: client/src/components/sign/components/login-form/login-form.jsx [hunk 1]
```diff
@@ -10,11 +10,10 @@ import {
 } from 'common/enums/enums';
 import {
   Button,
-  Form,
   FormInput,
-  Segment,
   Message,
-  NavLink
+  NavLink,
+  UiSegment
```
Comment (what1s1ove) [inline client/src/components/sign/components/login-form/login-form.jsx:16]:
```suggestion
  Segment
```

File: client/src/components/sign/components/registration-form/registration-form.jsx [hunk 2]
```diff
@@ -10,11 +10,10 @@ import {
 } from 'common/enums/enums';
 import {
   Button,
-  Form,
   FormInput,
-  Segment,
   Message,
-  NavLink
+  NavLink,
+  UiSegment
```
Comment (what1s1ove) [inline client/src/components/sign/components/registration-form/registration-form.jsx:16]:
```suggestion
  Segment
```

File: client/src/components/thread/components/add-comment/add-comment.jsx [hunk 3]
```diff
@@ -14,17 +16,23 @@ const AddComment = ({ postId, onCommentAdd }) => {
...
+    <>
+      <TextArea
+        className={styles.commentArea}
         value={body}
         placeholder="Type a comment..."
-        onChange={ev => setBody(ev.target.value)}
+        onChange={handleTextChange}
       />
-      <Button type={ButtonType.SUBMIT} isPrimary>
+      <Button type={ButtonType.BUTTON} onClick={handleAddComment} isPrimary>
```
Comment (what1s1ove) [inline client/src/components/thread/components/add-comment/add-comment.jsx:32]:
```suggestion
      <Button type={ButtonType.SUBMIT} isPrimary>
```
set listener on form on `onSubmit` event

File: client/src/components/thread/components/add-post/add-post.jsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import { useState } from 'hooks/hooks';
+import { useCallback, useState } from 'hooks/hooks';
 import PropTypes from 'prop-types';
 import { ButtonColor, ButtonType, IconName } from 'common/enums/enums';
-import { Button, Form, Image, Segment } from 'components/common/common';
+import { Button, Image, TextArea, UiSegment } from 'components/common/common';
```
Comment (what1s1ove) [inline client/src/components/thread/components/add-post/add-post.jsx:4]:
```suggestion
import { Button, Image, TextArea, Segment } from 'components/common/common';
```

**Review by what1s1ove — COMMENTED** [review]
don't forget to clear `readme`

File: client/src/components/common/button/button.jsx [hunk 5]
```diff
@@ -23,25 +25,26 @@ const Button = ({
   children
 }) => {
   const hasIcon = Boolean(iconName);
+  const btnType = useMemo(() => (checkIsValidBtnType(type) ? type : ButtonType.BUTTON), [type]);
```
[the remark this answers is not in the document]
Comment (PavelSukhinin) [reply client/src/components/common/button/button.jsx:28]: it's done for sake of validation

[the remark this answers is not in the document]
Comment (what1s1ove) [reply client/src/components/common/button/button.jsx:28]: for validation we should use propTypes and `oneOf` 🙏

**Review by ghost — APPROVED** [review]

File: client/src/components/common/image/image.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  isCircular: PropTypes.bool,
+  className: PropTypes.string,
+  height: PropTypes.string,
+  size: PropTypes.oneOf(Object.values(ImageSize)),
+  src: PropTypes.string.isRequired,
+  width: PropTypes.string
+};
+
+Image.defaultProps = {
+  isCentered: undefined,
```
Comment (Liza-Veis) [inline client/src/components/common/image/image.jsx:44]:
```suggestion
  isCentered: false,
```

**Review by what1s1ove — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**